app/views/map/_userLeaflet.html.erb

```diff
@@ -0,0 +1,31 @@
+<% if haslocation == true %>
+  <% unique_id = rand(100) %>
+  <style>
+    #map<%= unique_id %> {  height:300px; margin: 0; position: relative;}
+  </style>
+  <div class="leaflet-map" id="map<%= unique_id %>"></div>
+  <script>
+    
+    	var map = L.map('map<%= unique_id %>').setView([<%= @map_lat %>,<%= @map_lon %>], 7);
+    	L.tileLayer("//a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(map);
+
+    	<% if !(@map_blurred == "true") %>
+    	L.marker([<%= @map_lat %>, <%= @map_lon %>]).addTo(map).bindPopup("<a href='https://publiclab.org/profile/<%= @profile_user.username  %>'><%= @profile_user.username  %> </a>");
+      <% end %>
+  </script>
+
+<% else %>
+    <div id="map_template" style="position: relative; display: inline-block;">
+      <img src="/images/maptemplate.png" style="height:300px; margin: 0; position: relative;">
+      
+      <button type='button' class='btn btn-primary btn-lg' onclick='myFunction()' style="position: absolute;    position: absolute;top: 41% ; left: 15% ;"> <strong> Share your Location </strong> </button>
```

Comment:
Can we change this from `myFunction()` to `addLocation()`? And tidy up some of the spacing? 

Also, can the `maptemplate.png` be swapped for this map tile: https://a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/0/0/0.png ?

Finally, can we switch from `btn-primary` to `btn-default`?

Thank you, sorry, i should've mentioned this earlier. Almost there!

Reply:
Okay , changes done , thanks :)